lib/csv_query/database.rb

```diff
@@ -1,37 +1,62 @@
 require 'sqlite3'
+require 'csv_query/table'
 
 module CsvQuery
-  # Wraps a SQLite in-memory database with a single table named csv.
+  # Wraps a SQLite in-memory database with a single table named csv by default if you pass in only a csv to initialize.
+  # Don't pass in a CSV and use #create_table and #import_data_from_csv to to perform more complex multi-table loads.
   class Database
-    attr_reader :database
+    attr_accessor :sqlite_database
+    alias_method :database, :sqlite_database # TODO: Can be removed by repo maintainer at version upgrade?
 
-    def initialize(csv)
-      @database = SQLite3::Database.new(':memory:')
-      @columns = csv.headers
-      create_table(@columns)
+    def initialize(default_csv_to_create_table_for=nil)
+      @tables = []
+      @sqlite_database = SQLite3::Database.new(':memory:')
+
+      create_table(default_csv_to_create_table_for, 'csv') if !default_csv_to_create_table_for.nil?
     end
 
-    def import_data_from_csv(csv)
-      columns = csv.headers
+    def create_table(csv, table_name)
+      CsvQuery::Table.new(self, csv, table_name).create_table
+    end
 
-      sql = "INSERT INTO csv VALUES (#{(['?'] * columns.size).join(',')})"
-      statement = database.prepare(sql)
+    def import_data_from_csv(csv, table_name='csv')
+      CsvQuery::Table.new(self, csv, table_name).import_data_from_csv
+    end
 
-      csv.each do |row|
-        statement.execute(row.fields)
-      end
+    def load_csv(csv, table_name)
+      table = CsvQuery::Table.new(self, csv, table_name)
+      table.create_table
+      table.import_data_from_csv
     end
 
     # Returns the results of sql. First row of the resultset contains the column names
     def query(sql)
-      database.execute2(sql)
+      @sqlite_database.execute2(sql)
```

Comment:
Please use the `sqlite_database` accessor rather than using the instance variable directly.